### DISCUSSION TITLE
Using with Emscripten

### DISCUSSION THREAD

**Discussion by tbuchs** [Daninet/gmp-wasm#8] [Q&A] [answered]
Hi, I have found this great library - however, as I am trying to cross compile a huge c++ framework to the web using Emscripten I was wondering if it is possible to use this with emscripten? I can compile the basic gmp library to use it with Emscripten, but have the Problem that it doesn't support 'long long' datatype.

Thanks for your help :)

**Comment by Daninet** [comment, marked as answer]
It's possible to expose it through Emscripten but it's quite complicated to link the two wasm binaries properly. It's easier to take a look on how I managed to fix the 'long long' issue here: https://github.com/Daninet/gmp-wasm/blob/f418285eb2d22e5ebb1da75cd213458b9f47c3c4/binding/gmp/Dockerfile#L17
